### PR TITLE
getWindow(): The current window, or null if the activity is not visual.

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
As this can be null, we have to check this.
Reported several times on google play console

Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### CODE REVIEW

**Comment by tobiasKaminsky** [comment]
@AndyScherzinger ping :-)

**Comment by AndyScherzinger** [comment]
:+1: fine by me

**Comment by AndyScherzinger** [comment]
cc @mario for second review

File: src/main/java/com/owncloud/android/authentication/PassCodeManager.java [hunk 1]
```diff
@@ -65,10 +65,12 @@ public static PassCodeManager getPassCodeManager() {
     protected PassCodeManager() {}
 
     public void onActivityCreated(Activity activity) {
-        if (passCodeIsEnabled() || fingerprintIsEnabled()) {
-            activity.getWindow().addFlags(WindowManager.LayoutParams.FLAG_SECURE);
-        } else {
-            activity.getWindow().clearFlags(WindowManager.LayoutParams.FLAG_SECURE);
+        if (activity.getWindow() != null) {
```
Comment (mario) [inline src/main/java/com/owncloud/android/authentication/PassCodeManager.java]: Assign result of activity.getWindow to a variable at the same time you're checking and use that afterwards.

File: src/main/java/com/owncloud/android/ui/activity/ExternalSiteWebView.java [hunk 2]
```diff
@@ -66,12 +66,14 @@ protected void onCreate(Bundle savedInstanceState) {
         showSidebar = extras.getBoolean(EXTRA_SHOW_SIDEBAR);
 
         // show progress
-        getWindow().requestFeature(Window.FEATURE_PROGRESS);
+        if (getWindow() != null) {
```
Comment (mario) [inline src/main/java/com/owncloud/android/ui/activity/ExternalSiteWebView.java]: Same as above.

File: src/main/java/com/owncloud/android/authentication/PassCodeManager.java [hunk 3]
```diff
@@ -65,10 +65,12 @@ public static PassCodeManager getPassCodeManager() {
...
+            if (passCodeIsEnabled() || fingerprintIsEnabled()) {
+                activity.getWindow().addFlags(WindowManager.LayoutParams.FLAG_SECURE);
```
Comment (mario) [inline src/main/java/com/owncloud/android/authentication/PassCodeManager.java]: Use previously stored thing.

File: src/main/java/com/owncloud/android/ui/activity/FingerprintActivity.java [hunk 4]
```diff
@@ -95,17 +95,17 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.fingerprintlock);
 
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP && getWindow() != null) {
```
Comment (mario) [inline src/main/java/com/owncloud/android/ui/activity/FingerprintActivity.java]: Same.

File: src/main/java/com/owncloud/android/ui/activity/PassCodeActivity.java [hunk 5]
```diff
@@ -92,29 +92,31 @@
...
-        mPassCodeEditTexts[0] = (EditText) findViewById(R.id.txt0);
+        mPassCodeEditTexts[0] = findViewById(R.id.txt0);
         mPassCodeEditTexts[0].setTextColor(ThemeUtils.primaryColor());
         mPassCodeEditTexts[0].getBackground().setColorFilter(ThemeUtils.primaryColor(), PorterDuff.Mode.SRC_ATOP);
         mPassCodeEditTexts[0].requestFocus();
-        getWindow().setSoftInputMode(
-                android.view.WindowManager.LayoutParams.SOFT_INPUT_STATE_VISIBLE);
 
-        mPassCodeEditTexts[1] = (EditText) findViewById(R.id.txt1);
+        if (getWindow() != null) {
```
Comment (mario) [inline src/main/java/com/owncloud/android/ui/activity/PassCodeActivity.java]: Same.

File: src/main/java/com/owncloud/android/ui/activity/Preferences.java [hunk 6]
```diff
@@ -747,25 +747,31 @@ private String getAppVersion() {
...
+            ThemeUtils.setColoredTitle(actionBar, getString(R.string.actionbar_settings));
+            actionBar.setBackgroundDrawable(new ColorDrawable(ThemeUtils.primaryColor()));
+
+            Drawable backArrow = getResources().getDrawable(R.drawable.ic_arrow_back);
+            actionBar.setHomeAsUpIndicator(ThemeUtils.tintDrawable(backArrow, ThemeUtils.fontColor()));
         }
 
-        Drawable backArrow = getResources().getDrawable(R.drawable.ic_arrow_back);
-        actionBar.setHomeAsUpIndicator(ThemeUtils.tintDrawable(backArrow, ThemeUtils.fontColor()));
+        if (getWindow() != null) {
```
Comment (mario) [inline src/main/java/com/owncloud/android/ui/activity/Preferences.java]: Same.

File: src/main/java/com/owncloud/android/ui/dialog/CreateFolderDialogFragment.java [hunk 7]
```diff
@@ -103,7 +103,11 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
                 .setTitle(ThemeUtils.getColoredTitle(getResources().getString(R.string.uploader_info_dirname),
                         accentColor));
         Dialog d = builder.create();
-        d.getWindow().setSoftInputMode(LayoutParams.SOFT_INPUT_STATE_VISIBLE);
+
+        if (d.getWindow() != null) {
```
Comment (mario) [inline src/main/java/com/owncloud/android/ui/dialog/CreateFolderDialogFragment.java]: Same.

File: src/main/java/com/owncloud/android/ui/dialog/CredentialsDialogFragment.java [hunk 8]
```diff
@@ -104,7 +104,11 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
                 .setNegativeButton(R.string.common_cancel, this);
 
         Dialog d = authDialog.create();
-        d.getWindow().setSoftInputMode(LayoutParams.SOFT_INPUT_STATE_VISIBLE);
+
+        if (d.getWindow() != null) {
```
Comment (mario) [inline src/main/java/com/owncloud/android/ui/dialog/CredentialsDialogFragment.java]: Same.

File: src/main/java/com/owncloud/android/ui/dialog/RenameFileDialogFragment.java [hunk 9]
```diff
@@ -117,7 +117,11 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
                 .setTitle(ThemeUtils.getColoredTitle(getResources().getString(R.string.rename_dialog_title),
                         accentColor));
         Dialog d = builder.create();
-        d.getWindow().setSoftInputMode(LayoutParams.SOFT_INPUT_STATE_VISIBLE);
+
+        if (d.getWindow() != null) {
```
Comment (mario) [inline src/main/java/com/owncloud/android/ui/dialog/RenameFileDialogFragment.java]: Same.

File: src/main/java/com/owncloud/android/ui/dialog/SharePasswordDialogFragment.java [hunk 10]
```diff
@@ -102,7 +102,11 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
                .setNegativeButton(R.string.common_cancel, this)
                .setTitle(R.string.share_link_password_title);
         Dialog d = builder.create();
-        d.getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_VISIBLE);
+
+        if (d.getWindow() != null) {
```
Comment (mario) [inline src/main/java/com/owncloud/android/ui/dialog/SharePasswordDialogFragment.java]: And again :P

File: src/main/java/com/owncloud/android/utils/ThemeUtils.java [hunk 11]
```diff
@@ -287,8 +287,8 @@ public static void colorSnackbar(Context context, Snackbar snackbar) {
      * @param color            the color
      */
     public static void colorStatusBar(FragmentActivity fragmentActivity, @ColorInt int color) {
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
-            fragmentActivity.getWindow().setStatusBarColor(color);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP && fragmentActivity.getWindow() != null) {
```
Comment (mario) [inline src/main/java/com/owncloud/android/utils/ThemeUtils.java]: and again.

File: src/main/java/com/owncloud/android/authentication/PassCodeManager.java (hunk 1 above)
Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/authentication/PassCodeManager.java]: Can you explain this a bit?
getWindow() can be null if the activity is not visual, according to doc.
So even if it quite strange, but the user could minimize the app during activity creation and thus then result in a NPE.
At least this is how I can explain the NPE reported by google play...

**Comment by tobiasKaminsky** [comment]
I changed the code, so ready for second round of CR.

File: src/main/java/com/owncloud/android/ui/activity/Preferences.java [hunk 12]
```diff
@@ -747,25 +748,32 @@ private String getAppVersion() {
...
-            getWindow().getDecorView().findViewById(actionBarTitleId).
-                    setContentDescription(getString(R.string.actionbar_settings));
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+                window.setStatusBarColor(ThemeUtils.primaryDarkColor(this));
+            }
+
+            // For adding content description tag to a title field in the action bar
+            int actionBarTitleId = getResources().getIdentifier("action_bar_title", "id", "android");
+            View actionBarTitleView = window.getDecorView().findViewById(actionBarTitleId);
+            if (actionBarTitleView != null) {    // it's null in Android 2.x
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/ui/activity/Preferences.java]: couldn't this be shortened since we don't support anything before 4.0.x?

**Comment by AndyScherzinger** [comment]
:+1: fine by me, just one minor comment

**Comment by AndyScherzinger** [comment]
extra lint warning is fine and will be fixed via #2508

**Comment by AndyScherzinger** [comment]
rebased to revive this PR, cc @mario @ardevd

File: src/main/java/com/owncloud/android/ui/activity/ExternalSiteWebView.java [hunk 13]
```diff
@@ -69,7 +69,10 @@ protected void onCreate(Bundle savedInstanceState) {
         showSidebar = extras.getBoolean(EXTRA_SHOW_SIDEBAR);
 
         // show progress
-        getWindow().requestFeature(Window.FEATURE_PROGRESS);
+        Window window = getWindow();
+        if (window != null) {
```
Comment (mario) [inline src/main/java/com/owncloud/android/ui/activity/ExternalSiteWebView.java]: if (getWindow() != null)

Comment (mario) [reply src/main/java/com/owncloud/android/ui/activity/ExternalSiteWebView.java]: Ignore the comment.

**Comment by mario** [comment]
👍

**Pull request merged**